src/intrinsics/ecma262/ArrayPrototype.js

```diff
@@ -992,6 +992,8 @@ export default function(realm: Realm, obj: ObjectValue): void {
       if (thisArg) {
         args.push(thisArg);
       }
+      // We need to materialize here, but there's no way of doing it without
+      // havocing, which brings about even more issues. :/
```

Comment:
This comment creates FUD rather than enlightenment. Can you reference an issue that spells out the issue in detail, along with examples?

Reply:
I can make a repro as a separate issue.

Reply:
And then please rewrite this comment and mention the issue number. As it stands right now, it's an odd comment to have in the code base.